src/imitation/algorithms/adversarial/common.py

```diff
@@ -322,25 +337,31 @@ def train_disc(
             write_summaries = self._init_tensorboard and self._global_step % 20 == 0
 
             # compute loss
-            batch = self._make_disc_train_batch(
+            self._disc_opt.zero_grad()
+
+            batch_iter = self._make_disc_train_batches(
                 gen_samples=gen_samples,
                 expert_samples=expert_samples,
             )
-            disc_logits = self.logits_expert_is_high(
-                batch["state"],
-                batch["action"],
-                batch["next_state"],
-                batch["done"],
-                batch["log_policy_act_prob"],
-            )
-            loss = F.binary_cross_entropy_with_logits(
-                disc_logits,
-                batch["labels_expert_is_one"].float(),
-            )
+            for batch in batch_iter:
+                disc_logits = self.logits_expert_is_high(
+                    batch["state"],
+                    batch["action"],
+                    batch["next_state"],
+                    batch["done"],
+                    batch["log_policy_act_prob"],
+                )
+                loss = F.binary_cross_entropy_with_logits(
+                    disc_logits,
+                    batch["labels_expert_is_one"].float(),
+                )
+
+                # Renormalise the loss to be averaged over the whole
+                # batch size instead of the minibatch size.
+                loss *= self.demo_minibatch_size / self.demo_batch_size
```

Comment:
I think it is, since there are checks to make sure `len(expert_samples) == len(gen_samples) == self.demo_batch_size` and an extra check of `self.demo_batch_size % self.demo_minibatch_size == 0`. However, I can see it being confusing, may be worth adding a comment.

Reply:
OK, I agree this is fine, difference is adversarial trainer can choose how many rollouts it samples.